Source/CharacteristicNotificationManager.swift

```diff
@@ -75,24 +82,14 @@ class CharacteristicNotificationManager {
             }
     }
 
-    private func setNotifyValue(_ enabled: Bool, for characteristic: Characteristic) -> Single<Characteristic> {
-        let observable = delegateWrapper
-            .peripheralDidUpdateNotificationStateForCharacteristic
-            .filter { $0.0 == characteristic.characteristic }
-            .take(1)
-            .map { (_, error) -> Characteristic in
-                if let error = error {
-                    throw BluetoothError.characteristicNotifyChangeFailed(characteristic, error)
-                }
-                return characteristic
-        }
-        let setObservable: Observable<Characteristic> = .deferred { [weak self] in
+    private func setNotifyValue(_ enabled: Bool, for characteristic: Characteristic) -> Completable {
+        return .deferred { [weak self] in
+            if enabled {
+                let counter = self?.uuidToActiveObservablesCountMap[characteristic.uuid] ?? 0
+                self?.uuidToActiveObservablesCountMap[characteristic.uuid] = counter + 1
```

Comment:
I think that it should be also guarded by NSLock. 
Imagine scenario:
1. thread 1 is on line 89, after getting counter value (value==1)
2. thread 2 goes to line 58 and decrease counter (value==0)
3. thread 1 changes coun map on line 89 (value==2)
Finally we have count==2 and only 1 observable.